pkgs/development/python-modules/pyspread/default.nix

```diff
@@ -25,32 +18,46 @@ buildPythonPackage rec {
     sha256 = "117268cbeb225e0aca4c59381b7bdf5333bf95ca85c1db410a326a887a74f3da";
   };
 
-  propagatedBuildInputs = [ numpy wxPython matplotlib pycairo python-gnupg xlrd xlwt jedi pyenchant basemap pygtk ];
-  # Could also (optionally) add pyrsvg and python bindings for libvlc
+  requiredPythonDependencies = [
+    numpy
+    pyqt5
+  ];
+
+  optionalPythonDependencies = [
+    matplotlib
+    pyenchant
+  ];
+
+  pythonDependencies =
+    requiredPythonDependencies 
+    ++ optionalPythonDependencies;
```

Comment:
Why include the optional dependencies? Typically we do not include optional dependencies; users can choose themselves whether to add them when composing an environment.